draft-ietf-tls-esni.md

```diff
@@ -174,14 +179,32 @@ ClientHelloInner in an "encrypted_client_hello" extension, which this document
 defines ({{encrypted-client-hello}}). Finally, it sends ClientHelloOuter to the
 server.
 
-Upon receiving ClientHelloOuter, the client-facing server processes the
-encrypted ClientHelloInner and either terminates the connection (in Shared Mode)
-or forwards it to the backend server (in Split Mode).
+Upon receiving the ClientHelloOuter, the server takes one of the following
+actions:
+
+1. If the client-facing server does not support ECH, it ignores the
+   "encrypted_client_hello" extension and proceeds with the handshake as usual,
+   per {{RFC8446}}, Section 4.1.2.
+1. If the client-facing server supports ECH but cannot decrypt it, then it
+   ignores the extension and proceeds with the handshake as usual. This is
+   referred to as "ECH rejection". When ECH is rejected, the server sends an
+   acceptable ECH configuration in its EncryptedExtensions message.
+1. If the client-facing server supports ECH and can decrypt it, then it forwards
+   the CllientHelloInner to the backend, who terminates the connection. This is
+   referred to as "ECH acceptance".
+
+Upon receiving the server's response, the client must first determine whether
+the ClientHelloOuter or ClientHelloInner was used. [[TODO: Describe how this is
+done. Currently this is by trial decryption.]] Once it makes this determination,
+the client proceeds with the handshake, assuming the indicated ClientHello was
+used by the backend server.
+
+[[TODO: Give a high-level description of "ech_nonce".]]
```

Comment:
I don't think this is needed (here).

Reply:
As written, the "ech_nonce" section comes out of nowhere, since it's purpose isn't articulated until "security considerations". My view is that any code point defined in the spec needs to be mentioned in the overview. In fact, I would include "outer_extensions" here, too.

Reply:
Sure, but this isn't the right place for it. It's a tiny part of the protocol. We can describe inline below when it arises.

Reply:
Ack, moving TODO to section on ech_nonce.